geode-cq/src/distributedTest/java/org/apache/geode/cache/query/cq/CQMetricsDUnitTest.java

```diff
@@ -0,0 +1,169 @@
+  private QueryService qs;
+  private TestCqListener testListener;
+  private MemberVM locator, server1, server2;
+
+  @Rule
+  public ClusterStartupRule cluster = new ClusterStartupRule(5);
+
+  @Rule
+  public GfshCommandRule gfsh = new GfshCommandRule();
+
+  @Before
+  public void setUpServers() throws Exception {
+    locator = cluster.startLocatorVM(0, l -> l.withoutClusterConfigurationService());
+    server1 = cluster.startServerVM(1, locator.getPort());
+    server2 = cluster.startServerVM(2, locator.getPort());
+
+    ClientCache clientCache = createClientCache(locator.getPort());
+    Region region =
+        clientCache.createClientRegionFactory(ClientRegionShortcut.CACHING_PROXY).create("region");
+
+    qs = clientCache.getQueryService();
+    CqAttributesFactory cqaf = new CqAttributesFactory();
+    testListener = new TestCqListener();
+    cqaf.addCqListener(testListener);
+
+    cqa = cqaf.create();
+    gfsh.connectAndVerify(locator);
+  }
+
+  @Test
+  public void testStopCq() throws Exception {
+    gfsh.executeAndAssertThat("create region --name=region --type=PARTITION")
+        .statusIsSuccess();
+    qs.newCq("Select * from /region r where r.ID = 1", cqa).execute();
+
+    server1.invoke(() -> populateRegion(0, 100));
+
+    locator.invoke(() -> {
+      Cache cache = getCache();
+      ManagementService service = ManagementService.getManagementService(cache);
+      DistributedSystemMXBean dsmbean = service.getDistributedSystemMXBean();
+      await().untilAsserted(() -> assertThat(dsmbean.getActiveCQCount()).isEqualTo(2));
+    });
+
+    // stop cq
+    qs.stopCqs();
+
+    locator.invoke(() -> {
+      Cache cache = getCache();
+      ManagementService service = ManagementService.getManagementService(cache);
+      DistributedSystemMXBean dsmbean = service.getDistributedSystemMXBean();
+      await().untilAsserted(() -> assertThat(dsmbean.getActiveCQCount()).isEqualTo(0));
+    });
+
+    checkActiveCqCount(server1, 0);
+    checkActiveCqCount(server2, 0);
+  }
+
+  @Test
+  public void testCloseCq() throws Exception {
+    gfsh.executeAndAssertThat("create region --name=region --type=PARTITION")
+        .statusIsSuccess();
+    qs.newCq("Select * from /region r where r.ID = 1", cqa).execute();
+
+    server1.invoke(() -> populateRegion(0, 100));
+
+    locator.invoke(() -> {
+      Cache cache = getCache();
+      ManagementService service = ManagementService.getManagementService(cache);
+      DistributedSystemMXBean dsmbean = service.getDistributedSystemMXBean();
+      await().untilAsserted(() -> assertThat(dsmbean.getActiveCQCount()).isEqualTo(2));
+    });
+
+    // close cq
+    qs.closeCqs();
+
+    locator.invoke(() -> {
+      Cache cache = getCache();
+      ManagementService service = ManagementService.getManagementService(cache);
+      DistributedSystemMXBean dsmbean = service.getDistributedSystemMXBean();
+      await().untilAsserted(() -> assertThat(dsmbean.getActiveCQCount()).isEqualTo(0));
```

Comment:
Please provide a timeout. I tested it with `timeout(30, TimeUnit.SECONDS)`. This way, this test will terminate in a "reasonable" time.